### ISSUE TITLE
Recovered Kerbals go Lost/Missing

### ISSUE THREAD

**Issue by MrKiel** [mmoench/KSTS#3]
When using a Transport Mission, transporting Kerbals _from_ a craft results in the kerbals being briefly available at the astronaut complex, and then going missing. 

Here's the relevant log file entries:

```
[KSTS] recovered kerbonaut Neldorf Kerman from vessel Kerbin Space Station
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

[KSTS] recovered kerbonaut Lobrett Kerman from vessel Kerbin Space Station
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

<snip>

[ProtoCrewMember Warning]: Crewmember Lobrett Kerman found inside a part but status is set as missing. Vessel must have failed to save earlier. Restoring assigned status.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

[ProtoCrewMember Warning]: Crewmember Neldorf Kerman found inside a part but status is set as missing. Vessel must have failed to save earlier. Restoring assigned status.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

<snip>

[ProtoCrewMember Warning]: Crewmember Lobrett Kerman found assigned but no vessels reference him. Lobrett Kerman set as missing.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

[ProtoCrewMember Warning]: Crewmember Neldorf Kerman found assigned but no vessels reference him. Neldorf Kerman set as missing.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

```

Full log file attached. 

[output_log.txt](https://github.com/mmoench/KSTS/files/573027/output_log.txt)

**Comment by mmoench** [comment]
Hm, I see. I've been working on my old Life-Support mod lately (CLLS) and have hit the same issue there...
As far as I can tell, KSP is keeping a backup of the Kerbal's status somewhere and tries to restore it, just like the message says. Thus far I was unable to completely remove kerbals from a vessel in such a way that this does not occur, so I will probably have to implement a workaround which sets the kerbal back to available once KSP tries to "fix" him. This of course will need some rigorous testing. I already fear complications when a freshly recovered kerbal is assigned to a new vessel...

**Comment by mmoench** [comment]
I was unable to fix the underlying problem, but I've implemented a workaround in the 1.6 release which takes care of the issue: Right before saving a game the mod now checks all kerbals and maybe re-corrects their roster status which might have been wrongly "corrected" be KSP a few moments earlier.
